gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -183,6 +186,7 @@ public void onDeleteSpec(URI deletedSpecURI, String deletedSpecVersion, Properti
     _log.info("Spec deletion detected: " + deletedSpecURI + "/" + deletedSpecVersion);
 
     if (topologyCatalog.isPresent()) {
+      //todo: de-risk: flow spec and topology spec share same uri?
```

Comment:
this shouldn't be possible unless some user wants to explicitly name their flow config the url of their spec executor.

Reply:
yeah, that's what I'm worry about, if that happens, we will ends up with issue that affect all the flows. Might be better to add the info of spec type in to properties?

Reply:
It's highly unlikely but if this is a scenario we can add it as a guard during compilation